### PR TITLE
Traverse exemplars in a quasi-random order in ND aggregator

### PR DESCRIPTION
**Pull request by oleksiyskononenko**
When exemplars are traversed in the order they appear in the `exemplars` vector, first exemplars will always have a priority over the last ones, hence, first exemplars "absorb" most of the members. In this case one can not distinguish real outliers with the exemplars (the last ones) that just had no chances to get some members.

One of the ways to fix it is to traverse exemplars in a random order, so that each exemplar gets the "same" chances to "absorb" a particular member. Here we use modular arithmetic to traverse exemplars in quasi-random order, as this approach seems to have negligible overhead, when comparing to `std::random_shuffle`, and we do not really need to be fully random, but just different enough to not inflate some particular exemplars.

### CODE REVIEW

File: c/extras/aggregator.cc [hunk 1]
```diff
@@ -641,6 +667,25 @@ void Aggregator::group_nd(const dtptr& dt, dtptr& dt_members) {
 }
 
 
+void Aggregator::fill_coprimes(size_t n, std::vector<size_t>& coprimes) {
+  coprimes.clear();
+  for (size_t i = 1; i <= n; ++i) {
+    if (gcd(i, n) == 1) coprimes.push_back(i);
```
Comment (st-pasha) [inline c/extras/aggregator.cc]: Eratosthenes Sieve is a much better approach to finding coprimes:
- create a bitmask of length `n`, filled with 0s;
- starting with `i = 2`:
  - if `mask[i]` is 1 then continue;
  - if `n` is divisible by `i`, then set all elements of the bitmask that are multiples of `i` to 1;
- continue until `i*i < n`.
In the end, all elements of bitmask that are 0s are coprime to `n`.

Comment (oleksiyskononenko) [reply c/extras/aggregator.cc]: Yes, sure, current approach can be optimized, I was more concerned about the final `members_count` distribution for the moment. For the aggregation purpose we can even stick to the list of prime numbers that are smaller than the typical `nd_max_bins`, i.e. ~1000, and then just use these prime numbers as the "seed".

Comment (oleksiyskononenko) [reply c/extras/aggregator.cc]: Actually, it seems that we can just use any prime number that is larger than `nd_max_bins`.

Comment (oleksiyskononenko) [reply c/extras/aggregator.cc]: Yes, sorry. I probably kept in mind the approach of `[n / 2; n]` for the range of prime numbers.

Comment (oleksiyskononenko) [reply c/extras/aggregator.cc]: Ok, I've changed it to Eratosthenes sieve, stopping at `i == n / 2`. Could probably optimized even further, but still good for our purpose.

**Pull request merged**